minor fixes of some Web Audio API articles (#15694)
mdn/content: files/en-us/web/api/web_audio_api/basic_concepts_behind_web_audio_api/index.md

  - concepts
  - sound
---

> This article explains some of the audio theory behind how the features of the Web Audio API work, to help you make informed decisions while designing how audio is routed through your app. It won't turn you into a master sound engineer, but it will give you enough background to understand why the Web Audio API works like it does.

Reviewer: The original phrasing assumes that the reader is not a good sound engineer. Let's rephrase:
```suggestion
This article explains some of the audio theory behind how the features of the Web Audio API work, to help you make informed decisions while designing how audio is routed through your app. If you are not a sound engineer, it will give you enough background to understand why the Web Audio API works as it does.
```

Reviewer 2: this is a good clarification, but it might flow better as:
> If you are not _already_ a sound engineer [...]

Reviewer: Updated

A sample is a single 32-bit floating point value that represents the value of the audio stream at each specific point in time, in a specific channel (left or right, if in the case of stereo). A frame, or sample frame, is the set of all values for all channels that will play at a specific point in time: all the samples of all the channels that play at the same time (two for a stereo sound, six for 5.1, etc.).

The sample rate is the number of those samples (or frames, since all samples of a frame play at the same time) that will play in one second, measured in Hz. The higher the sample rate, the better the sound quality.

Let's look at a _mono_ and a _stereo_ audio buffer, each one second long at a rate of 44100Hz:

- The _mono_ buffer will have 44100 samples, and 44100 frames. The `length` property will be 44100.
- The _stereo_ buffer will have 88200 samples, but still 44100 frames. The `length` property will still be 44100 since it's equal to the number of frames.

![A diagram showing several frames in an audio buffer in a long line, each one containing two samples, as the buffer has two channels, it is stereo.](sampleframe-english.png)

> When a buffer plays, you will first hear the leftmost sample frame, then the one right next to it, then the next, _ad infinitum_ until the end of the buffer. In the case of stereo, you will hear both channels at the same time. Sample frames are very useful because they are independent of the number of channels and they represent time in a ideal way for doing precise audio manipulation.

Reviewer: If it stops at the end of the buffer, it isn't _ad infinitum_ 😄 
```suggestion
When a buffer plays, you will first hear the leftmost sample frame, then the one right next to it, then the next, _and so on_, until the end of the buffer. In the case of stereo, you will hear both channels at the same time. Sample frames are very useful because they are independent of the number of channels and they represent time in a ideal way for doing precise audio manipulation.
```
